toolchains/solidity/extension/src/gas-estimation.ts

```diff
@@ -0,0 +1,344 @@
+import { execSync, exec } from "child_process";
+import * as vscode from "vscode";
+
+type GasReport = {
+  average: bigint;
+  min?: bigint;
+  max?: bigint;
+  median?: bigint;
+};
+
+type Function = {
+  name: string,
+  line: number
+};
+
+type Report = Map<string, Map<string, GasReport>>;
+
+type ReportDecorators = Map<string, vscode.DecorationOptions[]>;
+
+function isForgeInstalled(): boolean {
```

Comment:
This function is called every time a file is opened or saved, this should be done only once on startup

Reply:
but it don't disable the extension so quite useless ?

Reply:
It is still expensive in performance, especially if the output does not change between the calls